scripts/sanitycheck

```diff
@@ -1175,6 +1175,8 @@ class MakeGenerator:
             generator = "Unix Makefiles"
             generator_cmd = "$(MAKE)"
             verb = "VERBOSE=1" if VERBOSE else ""
+        if options.cmake_only:
+            generator_cmd = "#" + generator_cmd
```

Comment:
instead of this, I would split the template MAKE_RULE_TMPL into 2 sections, one for cmake, the other for the generator and only use the cmake one with the --cmake-only option like so:

```diff
--- a/scripts/sanitycheck
+++ b/scripts/sanitycheck
@@ -1108,7 +1108,7 @@ class MakeGenerator:
 {goal}:
 """

-    MAKE_RULE_TMPL = """\t@echo sanity_test_{phase} {goal} >&2
+    CMAKE_RULE_TMPL = """\t@echo sanity_test_{phase} {goal} >&2
 \tcmake  \\
 \t\t-G"{generator}"\\
 \t\t-H{directory}\\
@@ -1118,6 +1118,9 @@ class MakeGenerator:
 \t\t-DEXTRA_LDFLAGS="{ldflags}"\\
 \t\t{args}\\
 \t\t>{logfile} 2>&1
+"""
+
+    MAKE_RULE_TMPL = """\t@echo sanity_test_{phase} {goal} >&2
 \t{generator_cmd} -C {outdir}\\
 \t\t{verb} {make_args}\\
 \t\t>>{logfile} 2>&1
@@ -1175,8 +1178,9 @@ class MakeGenerator:
             generator = "Unix Makefiles"
             generator_cmd = "$(MAKE)"
             verb = "VERBOSE=1" if VERBOSE else ""
+
         if options.cmake_only:
-            generator_cmd = "#" + generator_cmd
+            generator_cmd =  generator_cmd

         if phase == 'running':
             return MakeGenerator.MAKE_RULE_TMPL_RUN.format(
@@ -1189,20 +1193,29 @@ class MakeGenerator:
                 make_args=make_args
             )
         else:
-            return MakeGenerator.MAKE_RULE_TMPL.format(
+            output = MakeGenerator.CMAKE_RULE_TMPL.format(
                 generator=generator,
-                generator_cmd=generator_cmd,
                 phase=phase,
                 goal=name,
                 outdir=outdir,
                 cflags=cflags,
                 ldflags=ldflags,
                 directory=workdir,
-                verb=verb,
                 args=args,
                 logfile=logfile,
-                make_args=make_args
             )
+            if not options.cmake_only:
+                output += MakeGenerator.MAKE_RULE_TMPL.format(
+                    generator_cmd=generator_cmd,
+                    phase=phase,
+                    goal=name,
+                    outdir=outdir,
+                    verb=verb,
+                    logfile=logfile,
+                    make_args=make_args
+                )
+
+            return output

     def _get_rule_footer(self, name):
         return MakeGenerator.GOAL_FOOTER_TMPL.format(goal=name)
```

Reply:
Make a change along these lines.